### PR TITLE
Feature/6.3/tdi 37456 rebase

### PR DESCRIPTION
**Pull request by wwang-talend**

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/34/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/34/)

File: components-jdbc/src/main/java/org/talend/components/jdbc/module/DriverTable.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import static org.talend.daikon.properties.property.PropertyFactory.newProperty;
+
+import java.util.List;
+
+import org.apache.commons.lang3.reflect.TypeLiteral;
+import org.talend.components.api.properties.ComponentPropertiesImpl;
+import org.talend.daikon.properties.presentation.Form;
+import org.talend.daikon.properties.property.Property;
+
+// have to implement ComponentProperties, not good
```
Comment (gonchar-ivan) [inline components-jdbc/src/main/java/org/talend/components/jdbc/module/DriverTable.java]: We need to check this later. There should not be i18n issues.

I had similar case in Jira components:
https://github.com/Talend/components/blob/master/components-jira/src/main/java/org/talend/components/jira/JiraConnectionProperties.java#L24

File: components-jdbc/src/main/java/org/talend/components/jdbc/module/JDBCConnectionModule.java [hunk 2]
```diff
@@ -11,7 +14,7 @@
     public Property<String> jdbcUrl = PropertyFactory.newProperty("jdbcUrl").setRequired();
 
     // TODO use the right widget for it
```
Comment (gonchar-ivan) [inline components-jdbc/src/main/java/org/talend/components/jdbc/module/JDBCConnectionModule.java]: Don't forget about TODOs

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCRowSourceOrSink.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        return ValidationResult.OK;
+    }
+
+    protected static ValidationResult fillValidationResult(ValidationResult vr, Exception ex) {
+        if (vr == null) {
+            return null;
+        }
+
+        if (ex.getMessage() == null || ex.getMessage().isEmpty()) {
+            vr.setMessage(ex.toString());
```
Comment (gonchar-ivan) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCRowSourceOrSink.java]: Ideally, ValidationResult messages should also be i18n
Example,
https://github.com/Talend/components/blob/master/components-jira/src/main/java/org/talend/components/jira/runtime/JiraSourceOrSink.java#L111-L129
+
https://github.com/Talend/components/blob/master/components-jira/src/main/resources/org/talend/components/jira/runtime/JiraSourceOrSink.properties

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/reader/JDBCRowReader.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            querySchema = properties.main.schema.getValue();
+        }
+        return querySchema;
+    }
+
+    @Override
+    public boolean start() throws IOException {
+        // TODO need to adjust the key
+        if (container != null) {
+            container.setComponentData(container.getCurrentComponentId(), "QUERY", properties.sql.getValue());
```
Comment (gonchar-ivan) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/reader/JDBCRowReader.java]: Flow variable.
Just to not forget. We will need to adjust it later:
1. Add QUERY flow variable in Definition.getReturnProperties()
2. Update this according updated flow variable mechanism
(currently it is not supported)

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+        return Collections.unmodifiableList(successfulWrites);
+    }
+
+    @Override
+    public List<IndexedRecord> getRejectedWrites() {
+        return Collections.unmodifiableList(rejectedWrites);
+    }
+
+    private void handleSuccess(IndexedRecord input) {
+        successCount++;
```
Comment (gonchar-ivan) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java]: you can use 
result.successCount++; 
Seems successCount field is not necessarily

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+            }
+        } catch (SQLException | ClassNotFoundException e) {
+            throw new ComponentException(e);
+        }
+    }
+
+    public void write(Object datum) throws IOException {
+        result.totalCount++;
+
+        successfulWrites.clear();
```
Comment (gonchar-ivan) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java]: It is cleared per record. 
Is it correct?

File: components-jdbc/src/test/java/org/talend/components/jdbc/JDBCRowTestIT.java [hunk 7]
```diff
@@ -0,0 +1,638 @@
...
+        DBTestUtils.releaseResource(connectionInfo);
+    }
+
+    @Before
+    public void before() throws ClassNotFoundException, SQLException, Exception {
+        DBTestUtils.prepareTableAndData(connectionInfo);
+    }
+
+    @Test
+    public void test_basic_no_connector() throws Exception {
```
Comment (gonchar-ivan) [inline components-jdbc/src/test/java/org/talend/components/jdbc/JDBCRowTestIT.java]: Violated Java naming conventions

**Comment by gonchar-ivan** [comment]
Also check Dmytro's comments there 
https://github.com/Talend/components/pull/243

File: components-jdbc/src/main/java/org/talend/components/jdbc/module/JDBCConnectionModule.java (hunk 2 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/module/JDBCConnectionModule.java]: sure, will push tup team to add it.

File: components-jdbc/src/main/java/org/talend/components/jdbc/module/DriverTable.java (hunk 1 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/module/DriverTable.java]: agree, check it later

File: components-jdbc/src/test/java/org/talend/components/jdbc/JDBCRowTestIT.java (hunk 7 above)
Comment (wwang-talend) [inline components-jdbc/src/test/java/org/talend/components/jdbc/JDBCRowTestIT.java]: sometimes,  i like this way as it's more easy to write, if use the right one :
testBasicNoConnectorWithLongNameAWithLongNameBWithLongNameC

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java (hunk 5 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java]: will consider it later, thanks.

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java (hunk 6 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/writer/JDBCRowWriter.java]: yes, only one by one, no aggregate action

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/reader/JDBCRowReader.java (hunk 4 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/reader/JDBCRowReader.java]: Yes

File: components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCRowSourceOrSink.java (hunk 3 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/runtime/JDBCRowSourceOrSink.java]: Agree, do it later

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/49/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/49/)

**Pull request merged**